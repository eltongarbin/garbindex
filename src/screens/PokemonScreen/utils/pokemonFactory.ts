const defaultPokemon = {
  height: 6,
  id: 35,
  name: 'clefairy',
  sprites: {
    front_default:
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/35.png'
  },
  stats: [
    {
      base_stat: 35,
      name: 'speed',
      id: 6
    },
    {
      base_stat: 65,
      name: 'special-defense',
      id: 5
    },
    {
      base_stat: 60,
      name: 'special-attack',
      id: 4
    },
    {
      base_stat: 48,
      name: 'defense',
      id: 3
    },
    {
      base_stat: 45,
      name: 'attack',
      id: 2
    },
    {
      base_stat: 70,
      name: 'hp',
      id: 1
    }
  ],
  weight: 75,
  image:
    'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/35.png',
  abilitiesById: {
    '56': {
      id: 56,
      name: 'cute-charm'
    },
    '98': {
      id: 98,
      name: 'magic-guard'
    },
    '132': {
      id: 132,
      name: 'friend-guard'
    }
  },
  typesById: {
    '18': {
      id: 18,
      name: 'fairy'
    }
  },
  evolvedFrom: 'cleffa'
};

export function create(overrides: any = defaultPokemon) {
  return { ...defaultPokemon, ...overrides };
}
