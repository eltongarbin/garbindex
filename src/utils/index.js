export { default as history } from './history';
export { default as theme } from './theme';
