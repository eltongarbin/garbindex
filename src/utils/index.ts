export { default as history } from './history';
export { default as theme } from './theme';

export const extractParamId = (url: any) => {
  const [id] = url.match(/(\d+)(?!.*\d)/);
  return id;
};
