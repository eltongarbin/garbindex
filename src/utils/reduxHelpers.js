import { createAction } from 'redux-actions';
import { call, put } from 'redux-saga/effects';
import isEmpty from 'lodash.isempty';

export const createType = (stateKey, type) =>
  `${process.env.REACT_APP_NAME}/${stateKey}/${type}`;

export const createAsyncTypes = (stateKey, type) => ({
  REQUEST: `${createType(stateKey, type)}_REQUEST`,
  SUCCESS: `${createType(stateKey, type)}_SUCCESS`,
  FAILURE: `${createType(stateKey, type)}_FAILURE`
});

export const createAsyncActions = (asyncTypes) => ({
  request: createAction(asyncTypes.REQUEST),
  receive: createAction(asyncTypes.SUCCESS),
  error: createAction(asyncTypes.FAILURE)
});

export function* genericAsyncResolver(
  fnService,
  parameter,
  asyncActions,
  cbFormatData
) {
  try {
    const data = yield call(fnService, parameter);

    let dataFormatted;
    if (cbFormatData && !isEmpty(data)) {
      dataFormatted = yield cbFormatData(data);
    }

    yield put(asyncActions.receive(dataFormatted || data));
  } catch (error) {
    yield put(asyncActions.error(error));
  }
}
