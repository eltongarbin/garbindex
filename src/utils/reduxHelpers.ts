import { call, put } from 'redux-saga/effects';
import { isEmpty } from 'lodash-es';

export function* genericAsyncResolver(
  fnService: any,
  parameter: any,
  asyncActions: any,
  cbFormatData: any
) {
  try {
    const data = yield call(fnService, parameter);

    let dataFormatted;
    if (cbFormatData && !isEmpty(data)) {
      dataFormatted = yield cbFormatData(data);
    }

    yield put(asyncActions.success(dataFormatted || data));
  } catch (error) {
    yield put(asyncActions.failure(error));
  }
}
