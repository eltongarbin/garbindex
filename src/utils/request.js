import axios from 'axios';

const client = axios.create({
  baseURL: 'https://pokeapi.co/api/v2'
});

const request = function(options) {
  const onSuccess = function(response) {
    return response.data;
  };

  const onError = function(error) {
    return Promise.reject(error.response || error.message);
  };

  return client(options)
    .then(onSuccess)
    .catch(onError);
};

export default request;
