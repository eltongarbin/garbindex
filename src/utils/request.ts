import axios from 'axios';

const client = axios.create({
  baseURL: process.env.REACT_APP_BASE_API_URL
});

const request = function(options: any) {
  const onSuccess = function(response: any) {
    return response.data;
  };

  const onError = function(error: any) {
    return Promise.reject(error.response || error.message);
  };

  return client(options)
    .then(onSuccess)
    .catch(onError);
};

export default request;
