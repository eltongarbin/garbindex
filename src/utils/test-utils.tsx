// test-utils.js
import React from 'react';
import { Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';

export function renderWithRouter(
  ui: React.ReactElement,
  {
    route,
    history = createMemoryHistory({ initialEntries: [route] })
  }: any = {}
) {
  const Wrapper = ({ children }: any) => (
    <Router history={history}>{children}</Router>
  );

  return {
    ...render(ui, { wrapper: Wrapper }),
    history
  };
}

// re-export everything
export * from '@testing-library/react';
