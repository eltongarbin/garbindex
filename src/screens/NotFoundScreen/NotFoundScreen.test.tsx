import React from 'react';
import { createMemoryHistory } from 'history';

import { renderWithRouter, fireEvent } from 'utils/test-utils';
import ScreenNotFound from './NotFoundScreen';

it('should render correctly', () => {
  const { container } = renderWithRouter(<ScreenNotFound />);
  expect(container).toMatchSnapshot();
});

it('should return to the previous location', () => {
  const history = createMemoryHistory();
  history.push('/previous-route');
  history.push('/actual-route');
  const { getByText } = renderWithRouter(<ScreenNotFound />, { history });

  expect(history.location.pathname).toBe('/actual-route');
  fireEvent.click(getByText('Voltar'));
  expect(history.location.pathname).toBe('/previous-route');
});
