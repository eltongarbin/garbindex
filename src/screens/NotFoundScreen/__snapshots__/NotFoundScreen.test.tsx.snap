// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-AxjAm hYgLoi"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root sc-AxirZ jgEvOh MuiTypography-h6 MuiTypography-colorInherit"
        >
          Page not found (404)
        </h6>
      </div>
    </header>
    <div
      class="MuiGrid-root sc-AxhCb cNYLIj MuiGrid-container MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-6"
      >
        <div
          class="sc-AxhUy binLdG"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Sorry, this page isn't here.
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Voltar
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
